### DISCUSSION TITLE
Mesh independence study and postprocessng of plotfiles

### DISCUSSION THREAD

**Discussion by JayStandridge** [AMReX-Combustion/PeleLMeX#360] [Q&A]
I was wondering how you perform mesh independence study with the Pele-suite of codes. I have converged a 2D axially symmetric laminar flame at 3 different refinement levels and want to verify the convergence of the solution. I usually user paraview for data visualization, but the filters would allow me to combine datasets to investigate differences in profiles are not supported for AMReX data. I know paraview supports HDF5 files, but it is not clear whether the AMReX capability to output HDF5 files is available for PeleC/PeleLMeX. It seems that yt (especially with https://github.com/nickwimer/ytScripts as a starting point) should be able to do a mesh independence study. Before I learned how to use yt, I thought it would be useful to ask how you verify mesh independence and identify what AMR parameters are sufficient for a given problem.

Thank you.

**Comment by baperry2** [comment]
The AMReX HDF5 capability is available in both PeleLMeX and PeleC. To use it in LMeX, compile with `USE_HDF5 = TRUE` and `HDF5_HOME = /path/to/hdf5` and then run with `amr.use_hdf5_plt = true`. 

For mesh independence with AMR, our typical approach is to add levels of refinement until whatever QOI we're interested in for that particular simulation converges - how much is "enough" resolution can depend on what your QOI is (could be a global quantity like flamespeed or local profiles). There is a bit of fuzziness - for example this assumes that your refinement criteria are adding resolution to the important areas of the flow. For some flows or single level computations, it makes more sense to refine the base level. You can see some examples here: https://amrex-combustion.github.io/PeleLMeX/manual/html/Validation.html. You can also see how we rigorously verify the order of convergence in our testing suite: https://github.com/AMReX-Combustion/PeleLMeX/blob/development/.github/workflows/convergenceTesting.yml

**Reply by cjlauer16** [reply]
Should I use those same lines to compile and run PeleC, or are they different than LMeX? I followed those steps, but it doesn't output .h5 files, and I see 'Unused ParmParse Variables: [TOP]::amr.use_hdf5_plt(nvals = 1)  :: [true]'  at the end of the output file.

**Reply by baperry2** [reply]
For PeleC, the relevant input file keywords are:
```
pelec.write_hdf5_plots = true
pelec.hdf5_compression = None@0 #specify ZFP compression
```
